bot/constants.py

```diff
@@ -1,69 +1,204 @@
+            _recursive_update(original[key], new[key])
+        else:
+            original[key] = new[key]
+
+
+if Path("config.yml").exists():
+    log.info("Found `config.yml` file, loading constants from it.")
+    with open("config.yml") as f:
+        user_config = yaml.safe_load(f)
+    _recursive_update(_CONFIG_YAML, user_config)
+
+
+class YAMLGetter(type):
+    """
+    Implements a custom metaclass used for accessing
+    configuration data by simply accessing class attributes.
+    Supports getting configuration from up to two levels
+    of nested configuration through `section` and `subsection`.
+
+    `section` specifies the YAML configuration section (or "key")
+    in which the configuration lives, and must be set.
+
+    `subsection` is an optional attribute specifying the section
+    within the section from which configuration should be loaded.
+
+    Example Usage:
+
+        # config.yml
+        bot:
+            prefixes:
+                direct_message: ''
+                guild: '!'
+
+        # config.py
+        class Prefixes(metaclass=YAMLGetter):
+            section = "bot"
+            subsection = "prefixes"
+
+        # Usage in Python code
+        from config import Prefixes
+        def get_prefix(bot, message):
+            if isinstance(message.channel, PrivateChannel):
+                return Prefixes.direct_message
+            return Prefixes.guild
+    """
+
+    subsection = None
+
+    def __getattr__(cls, name):
+        name = name.lower()
+
+        try:
+            if cls.subsection is not None:
+                return _CONFIG_YAML[cls.section][cls.subsection][name]
+            return _CONFIG_YAML[cls.section][name]
+        except KeyError:
+            dotted_path = '.'.join(
+                (cls.section, cls.subsection, name)
+                if cls.subsection is not None else (cls.section, name)
+            )
+            log.critical(f"Tried accessing configuration variable at `{dotted_path}`, but it could not be found.")
+            raise
+
+    def __getitem__(cls, name):
+        return cls.__getattr__(name)
+
+
+class Bot(metaclass=YAMLGetter):
+    section = "bot"
+
+
+class Cooldowns(metaclass=YAMLGetter):
+    section = "bot"
+    subsection = "cooldowns"
+
+
+class Emojis(metaclass=YAMLGetter):
+    section = "bot"
+    subsection = "emojis"
+
+
+class Channels(metaclass=YAMLGetter):
+    section = "guild"
+    subsection = "channels"
+
+
+class Roles(metaclass=YAMLGetter):
+    section = "guild"
+    subsection = "roles"
+
+
+class Guild(metaclass=YAMLGetter):
+    section = "guild"
+
+
+class Keys(metaclass=YAMLGetter):
+    section = "keys"
+
+
+class ClickUp(metaclass=YAMLGetter):
+    section = "clickup"
+
+
+class Papertrail(metaclass=YAMLGetter):
+    section = "papertrail"
+
+
+class URLs(metaclass=YAMLGetter):
+    section = "urls"
+
 
 # Debug mode
 DEBUG_MODE = True if 'local' in os.environ.get("SITE_URL", "local") else False
```

Comment:
actually I'm not sure if we're even using this constant anymore. could you check and just remove it if not?

Reply:
Pretty sure we are:
```sh
$ grep -r 'DEBUG_MODE' .
./bot/constants.py:DEBUG_MODE = True if 'local' in os.environ.get("SITE_URL", "local") else False
./bot/converters.py:from bot.constants import DEBUG_MODE, Keys, URLs
./bot/converters.py:        if DEBUG_MODE:
./bot/cogs/logging.py:from bot.constants import Channels, DEBUG_MODE
./bot/cogs/logging.py:        if not DEBUG_MODE:
./bot/cogs/events.py:    Channels, DEBUG_MODE, Guild,
./bot/cogs/events.py:                if not DEBUG_MODE:
```

Reply:
fair enough.